controllers/news-controllers.js

```diff
@@ -50,8 +50,28 @@ exports.getCommentsByArticleId = (request, response, next) => {
         })
         .catch((error) => {
             if (error.code === '22P02') {
-                return response.status(400).send({ msg: 'Invalid article ID' });
+                response.status(400).send({ msg: 'Invalid article ID' });
             }
             next(error);
         })
 }
+
+exports.createComment = (request, response, next) => {
+    const { article_id } = request.params;
+    const { body, username } = request.body;
+
+    if(!body || !username) {
+        response.status(400).send({ msg: 'Missing required fields'})
+    }
+        
+    addComment(article_id, { body, author: username })
+    .then((comment) => {
+        response.status(201).send({ comment})
+    })
+    .catch((error) => {
+        if (error.code === '23503') {
```

Comment:
this should be in the custom error handlers and here you should just be passing your error to the next middleware